common/src/main/java/com/wynntils/wynn/model/PlayerModel.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model;
+
+import com.google.gson.JsonObject;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Model;
+import com.wynntils.core.net.ApiResponse;
+import com.wynntils.core.net.UrlId;
+import com.wynntils.mc.event.PlayerJoinedWorldEvent;
+import com.wynntils.mc.event.PlayerTeamEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.objects.WorldState;
+import com.wynntils.wynn.objects.account.AccountType;
+import com.wynntils.wynn.objects.account.WynntilsUser;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.scores.PlayerTeam;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public final class PlayerModel extends Model {
+    private static final Pattern GHOST_WORLD_PATTERN = Pattern.compile("^_(\\d+)$");
+
+    private final Map<UUID, WynntilsUser> users = new ConcurrentHashMap<>();
+    private final Set<UUID> fetching = ConcurrentHashMap.newKeySet();
+    private final Map<UUID, Integer> ghosts = new ConcurrentHashMap<>();
+
+    // Returns true if the player is on the same server and is not a npc
+    public boolean isLocalPlayer(Player player) {
+        return !isNpc(player) && !(isPlayerGhost(player));
+    }
+
+    public boolean isNpc(Player player) {
+        String scoreboardName = player.getScoreboardName();
+        return isNpc(scoreboardName);
+    }
+
+    public boolean isPlayerGhost(Player player) {
+        return ghosts.containsKey(player.getUUID());
+    }
+
+    public WynntilsUser getUser(UUID uuid) {
+        return users.getOrDefault(uuid, null);
+    }
+
+    @SubscribeEvent
+    public void onWorldStateChange(WorldStateEvent event) {
+        switch (event.getNewState()) {
+            case NOT_CONNECTED, CONNECTING -> clearUserCache();
+        }
+        if (event.getNewState() == WorldState.WORLD) {
+            clearGhostCache();
+        }
```

Comment:
Why not include this in the switch above?

Reply:
Our hades info needs to be updated only when connecting to the wynncraft network, but ghost data becomes irrelevant as soon as you switch to another world.